services/cd-service/pkg/repository/repository.go

```diff
@@ -2517,113 +2517,121 @@ func (s *State) UpdateEnvironmentsInOverview(ctx context.Context, transaction *s
 			var groupName = mapper.DeriveGroupName(config, envName)
 			var envInGroup = getEnvironmentInGroup(result.EnvironmentGroups, groupName, envName)
 
-			argocd := &api.EnvironmentConfig_ArgoCD{
-				SyncWindows: []*api.EnvironmentConfig_ArgoCD_SyncWindows{},
-				Destination: &api.EnvironmentConfig_ArgoCD_Destination{
-					Name:                 "",
-					Server:               "",
-					Namespace:            nil,
-					AppProjectNamespace:  nil,
-					ApplicationNamespace: nil,
-				},
-				AccessList:             []*api.EnvironmentConfig_ArgoCD_AccessEntry{},
-				ApplicationAnnotations: map[string]string{},
-				IgnoreDifferences:      []*api.EnvironmentConfig_ArgoCD_IgnoreDifferences{},
-				SyncOptions:            []string{},
-			}
-			if config.ArgoCd != nil {
-				argocd = mapper.TransformArgocd(*config.ArgoCd)
-			}
-			env := api.Environment{
-				DistanceToUpstream: 0,
-				Priority:           api.Priority_PROD,
-				Name:               envName,
-				Config: &api.EnvironmentConfig{
-					Upstream:         mapper.TransformUpstream(config.Upstream),
-					Argocd:           argocd,
-					EnvironmentGroup: &groupName,
+			err2 := s.UpdateEnvironmentInternal(ctx, transaction, config, envName, groupName, envInGroup)
```

Comment:
We're still recalculating the whole overview. Shouldn't we put a filter to update only the created environment?

Reply:
yes, but that's more complicated than i thought, so i split off this PR.